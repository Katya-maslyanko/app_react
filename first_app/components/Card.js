// Импортируем необходимые модули и компоненты
import React from "react";
import { useNavigation } from "@react-navigation/native";
import { View, Image } from "react-native";
import { COLORS, SIZES, SHADOWS } from "../constants";
import { SubInfo, Title } from "./SubInfo";
import { RectButton } from "./Button";

const Card = ({ data }) => {
  // Получаем объект навигации из React Navigation
  const navigation = useNavigation();

  return (
    <View
      style={{
        backgroundColor: COLORS.white,
        borderRadius: SIZES.font,
        marginBottom: SIZES.extraLarge,
        margin: SIZES.base,
        ...SHADOWS.dark,
      }}
    >
      {/* Контейнер для изображения */}
      <View
        style={{
          width: "100%",
          height: 100,
        }}
      >
        <Image
          source={data.image}
          resizeMode="cover"
          style={{
            width: "100%",
            height: "100%",
            borderTopLeftRadius: SIZES.font,
            borderTopRightRadius: SIZES.font,
          }}
        />
      </View>

      {/* Компонент для отображения дополнительной информации */}
      <SubInfo />

      <View style={{ width: "100%", padding: SIZES.font }}>
        {/* Компонент для отображения названия и создателя  */}
        <Title
          title={data.name}
          subTitle={data.creator}
          titleSize={SIZES.large}
          subTitleSize={SIZES.small}
        />

        <View
          style={{
            marginTop: SIZES.font,
            flexDirection: "row",
            justifyContent: "space-between",
            alignItems: "center",
            justifyContent: "flex-end",
          }}
        >
          {/* Кнопка для перехода на экран деталей  */}
          <RectButton
            minWidth={120}
            fontSize={SIZES.font}
            handlePress={() => navigation.navigate("Details", { data })}
          />
        </View>
      </View>
    </View>
  );
};

export default Card;