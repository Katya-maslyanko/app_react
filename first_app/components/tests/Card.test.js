// Импорт необходимых модулей и библиотек
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import Card from '../Card';

// Описание тестового блока для компонента Card
describe('Card component', () => {
  // Описание конкретного теста, проверяющего правильное отображение компонента
  it('Правильное отображение компонента', () => {
    // Создание тестовых данных для компонента Card
    const data = {
      image: 'https://example.com/image.png',
      name: 'Стоимость разработки мобильного приложения',
      creator: 'Команда «Иностудио»',
    };

    // Рендеринг компонента Card внутри NavigationContainer
    const { toJSON } = render(
      <NavigationContainer>
        <Card data={data} />
      </NavigationContainer>
    );

    // Проверка, что результат рендеринга соответствует ожидаемому снимку (snapshot)
    expect(toJSON()).toMatchSnapshot();
  });
});