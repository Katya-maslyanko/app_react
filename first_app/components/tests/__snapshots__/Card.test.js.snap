// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Правильное отображение компонента 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "borderRadius": 14,
      "elevation": 14,
      "margin": 8,
      "marginBottom": 24,
      "shadowColor": "#74858C",
      "shadowOffset": {
        "height": 7,
        "width": 0,
      },
      "shadowOpacity": 0.41,
      "shadowRadius": 9.11,
    }
  }
>
  <View
    style={
      {
        "height": 100,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source="https://example.com/image.png"
      style={
        {
          "borderTopLeftRadius": 14,
          "borderTopRightRadius": 14,
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": -24,
        "paddingHorizontal": 14,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          {
            "testUri": "../../../assets/images/person02.png",
          }
        }
        style={
          {
            "height": 48,
            "marginLeft": 0,
            "width": 48,
          }
        }
      />
      <Image
        resizeMode="contain"
        source={
          {
            "testUri": "../../../assets/images/person03.png",
          }
        }
        style={
          {
            "height": 48,
            "marginLeft": -14,
            "width": 48,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 14,
          "elevation": 1,
          "justifyContent": "center",
          "maxWidth": "50%",
          "paddingHorizontal": 14,
          "paddingVertical": 8,
          "shadowColor": "#74858C",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
        }
      }
    >
      <Text
        style={
          {
            "color": "#003153",
            "fontFamily": "InterRegular",
            "fontSize": 12,
          }
        }
      >
        Читать
      </Text>
      <Text
        style={
          {
            "color": "#003153",
            "fontFamily": "InterSemiBold",
            "fontSize": 16,
          }
        }
      >
        1 минута
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "padding": 14,
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "#003153",
            "fontFamily": "InterSemiBold",
            "fontSize": 18,
          }
        }
      >
        Стоимость разработки мобильного приложения
      </Text>
      <Text
        style={
          {
            "color": "#003153",
            "fontFamily": "InterRegular",
            "fontSize": 12,
          }
        }
      >
        Писатель: 
        Команда «Иностудио»
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginTop": 14,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#003153",
            "borderRadius": 24,
            "minWidth": 120,
            "opacity": 1,
            "padding": 12,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFF",
              "fontFamily": "InterSemiBold",
              "fontSize": 14,
              "textAlign": "center",
            }
          }
        >
          Перейти
        </Text>
      </View>
    </View>
  </View>
</View>
`;
